src/power_grid_model_io/converters/pandapower_converter.py

```diff
@@ -0,0 +1,518 @@
+                    extra_info[pgm_idx] = {"table": pp_table, "index": pp_idx}
+
+        return self.pgm_data
+
+    def _serialize_data(self, data: Dataset, extra_info: Optional[ExtraInfoLookup] = None) -> PandasData:
+        raise NotImplementedError()
+
+    def _create_input_data(self):
+        self._create_pgm_input_nodes()
+        self._create_pgm_input_lines()
+        self._create_pgm_input_sources()
+        self._create_pgm_input_sym_loads()
+        self._create_pgm_input_shunts()
+        self._create_pgm_input_transformers()
+        self._create_pgm_input_sym_gens()
+        self._create_pgm_input_three_winding_transformers()
+        self._create_pgm_input_links()
+
+    def _create_pgm_input_nodes(self):
+        assert "node" not in self.pgm_data
+
+        pp_busses = self.pp_data["bus"]
+        pgm_nodes = initialize_array(data_type="input", component_type="node", shape=len(pp_busses))
+        pgm_nodes["id"] = self._generate_ids("bus", pp_busses.index)
+        pgm_nodes["u_rated"] = pp_busses["vn_kv"] * 1e3
+
+        self.pgm_data["node"] = pgm_nodes
+
+    def _create_pgm_input_lines(self):
+        assert "line" not in self.pgm_data
+
+        pp_lines = self.pp_data["line"]
+
+        switch_states = self.get_switch_states("line")
+
+        pgm_lines = initialize_array(data_type="input", component_type="line", shape=len(pp_lines))
+        pgm_lines["id"] = self._generate_ids("line", pp_lines.index)
+        pgm_lines["from_node"] = self._get_ids("bus", pp_lines["from_bus"])
+        pgm_lines["from_status"] = pp_lines["in_service"] & switch_states.iloc[0, :]
+        pgm_lines["to_node"] = self._get_ids("bus", pp_lines["to_bus"])
+        pgm_lines["to_status"] = pp_lines["in_service"] & switch_states.iloc[1, :]
+        pgm_lines["r1"] = pp_lines["r_ohm_per_km"] * pp_lines["length_km"] / pp_lines["parallel"]
+        pgm_lines["x1"] = pp_lines["x_ohm_per_km"] * pp_lines["length_km"] / pp_lines["parallel"]
+        pgm_lines["c1"] = pp_lines["c_nf_per_km"] * pp_lines["length_km"] * pp_lines["parallel"] * 1e-9
+
+        # The formula for tan1 = R_1 / Xc_1 = (g * 1e-6) / (2 * pi * f * c * 1e-9) = g / (2 * pi * f * c * 1e-3)
+        pgm_lines["tan1"] = pp_lines["g_us_per_km"] / (
+            2 * np.pi * self.system_frequency * pp_lines["c_nf_per_km"] * 1e-3
+        )
+
+        pgm_lines["i_n"] = (pp_lines["max_i_ka"] * 1e3) * pp_lines["df"] * pp_lines["parallel"]
+
+        self.pgm_data["line"] = pgm_lines
+
+    def _create_pgm_input_sources(self):
+        assert "source" not in self.pgm_data
+
+        pp_ext_grid = self.pp_data["ext_grid"]
+
+        pgm_sources = initialize_array(data_type="input", component_type="source", shape=len(pp_ext_grid))
+        pgm_sources["id"] = self._generate_ids("ext_grid", pp_ext_grid.index)
+        pgm_sources["node"] = self._get_ids("bus", pp_ext_grid["bus"])
+        pgm_sources["status"] = pp_ext_grid["in_service"]
+        pgm_sources["u_ref"] = pp_ext_grid["vm_pu"]
```

Comment:
Also convert `va_degree` for `u_angle_ref` in pgm.

Also convert `rx_ratio` and `z01_ratio` in pgm.